doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+# Error Message Concept
+
+The purpose of this file is solely to better comment on in a Pull Request and gather feedback.
+
+## Findings/ Motivation
+
+1. Too many errors in the [specification](src\error\specification) (210 errors/warnings)
+2. Many duplicate or redundant errors
+3. Description/ Reason are very similar
+4. Potential file splitting (1327 LOC)
+5. Verbose error message
+
```

Comment:
Another problem is that strerror sometimes is not part of the reason, sometimes (not) in quotes, sometimes \n is at the end. (These are basically bugs to be fixed but maybe a better error concept would also avoid such errors.)

Reply:
I do not think that such problem is solvable here as in most languages exceptions/error messages are getting simply converted into another one (encapsulation, layering, etc.) and this can simply lead to information loss or formatting problems. It is up to the developer to do it correctly in my opinion.

Reply:
I could just think about another method which takes the strerror() as a parameter and the error logging tool handles it

Reply:
Another method will not help much. The framework should reset errno correctly and it should be part of the error/warning if it was changed by the plugin.

We should also have tests if errno is handled correctly.